src/lib/comms/sol-http-client-impl-curl.c

```diff
@@ -768,9 +672,20 @@ static bool
 set_post_fields_from_params(CURL *curl, struct sol_arena *arena,
     const struct sol_http_param *params)
 {
-    char *post = build_post_fields(curl, params);
-
-    return set_string_option(curl, CURLOPT_POSTFIELDS, arena, post);
+    struct sol_buffer buf;
+    char *post;
+    int err;
+    bool r = true;
+
+    sol_buffer_init(&buf);
+    err = sol_http_encode_params(&buf, SOL_HTTP_PARAM_POST_FIELD, params);
+    SOL_INT_CHECK(err, < 0, false);
+
+    post = sol_buffer_steal_or_copy(&buf, NULL);
+    if (post)
+        r = set_string_option(curl, CURLOPT_POSTFIELDS, arena, post);
```

Comment:
change this to `CURLOPT_COPYPOSTFIELDS`, see the comment above about `set_string_option()` usage.